devito/ir/clusters/queue.py

```diff
@@ -1,13 +1,16 @@
+from collections import defaultdict
 from itertools import groupby
 
-__all__ = ['Queue']
+from devito.ir.support import Scope
+from devito.tools import as_tuple, flatten
+
+__all__ = ['Queue', 'QueueStateful']
 
 
 class Queue(object):
 
     """
-    A special queue to process objects in nested IterationSpaces based on
-    a divide-and-conquer algorithm.
+    A special queue to process Clusters based on a divide-and-conquer algorithm.
```

Comment:
Queue?

Reply:
it's not a queue, it's basically a special "implicit-tree visitor". This is just a legacy name. I'll change it to something else once and for all in another PR, but before another minor cleanup will be required, so I'm sticking to Queue for the time being.

It's not "too wrong" in the sense that it processes clusters in FIFO order (like a standard queue), where FIFO is dictated by how the clusters would appear in the generated code